### PR TITLE
Chore: replace `react-popper` with `floating-ui` in `Popover`

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- replace `react-popper` with `floating-ui` in `Popover`
- `Popover` is used in a few places:
  - `ColorPicker`
    - this can be tested directly in storybook
    - only real change is the offset has changed from 15px (which was custom css in the `ColorPicker` component) to 8 (default set in `Popover` component)
  - `ResourcePicker`
    - this can be tested in the canvas panel. add an icon element, then change the icon svg. the picker that appears uses `Popover`.
    - again, i think the only thing that has changed is the offset 🤔 
  - `Table` -> `Filter`
    - this can be tested in storybook by adding `filterable: true` to the custom section 
  - `HoverCard` in alerting
    - verified this is ok
  - `SpanBar` in traces
    - can be tested by going to `Explore` -> `TestData` -> `Trace` scenario and hovering the bars 
    - verified this is ok

**Why do we need this feature?**

- `@floating-ui/react` is the next evolution of popper
- better api 😍 
- help to standardise on `@floating-ui/react`

**Who is this feature for?**

- grafana devs

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/Tooltip/Popover.tsx [hunk 1]
```diff
@@ -1,96 +1,102 @@
...
 interface Props extends Omit<React.HTMLAttributes<HTMLDivElement>, 'content'> {
   show: boolean;
-  placement?: Placement;
+  placement?: TooltipPlacement;
   content: PopoverContent;
-  referenceElement: HTMLElement | VirtualElement;
+  referenceElement: HTMLElement;
   wrapperClassName?: string;
-  renderArrow?: RenderPopperArrowFn;
+  renderArrow?: boolean;
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/Tooltip/Popover.tsx:25]: i guess this is a breaking change? 🤔

Comment (eledobleefe) [reply packages/grafana-ui/src/components/Tooltip/Popover.tsx:25]: According to the next comment, I think so 😅

**Review by eledobleefe — APPROVED** [review]
According to my little knowledge, LGTM 👍

File: packages/grafana-ui/src/components/Tooltip/Popover.tsx (hunk 1 above)
Comment (joshhunt) [reply packages/grafana-ui/src/components/Tooltip/Popover.tsx:25]: I actually don't think this is a (runtime) breaking change. JSX will be coerced to boolean and render the default floating-ui arrow if specified.

**Review by joshhunt — APPROVED** [review]

**Pull request merged**